### PR TITLE
new dev docs

### PR DESCRIPTION
**Pull request by lucksp**
# Description
add `switchWidget` and update `setWidget`

### Related Issue(s)
- https://snapengage.atlassian.net/browse/DEV-6675

# Ready-to-Merge Checklist
- [ ] Adds or updates dependencies?
- [ ] Have you built these changes locally?

### CODE REVIEW

File: source/assets/stylesheets/screen.css.scss [hunk 1]
```diff
@@ -494,6 +494,12 @@ body {
   }
 }
 
+.depracated {
```
Comment (sejanick) [inline source/assets/stylesheets/screen.css.scss]: This should be spelled "deprecated".

File: source/includes/_setwidgetid.md [hunk 2]
```diff
@@ -1,11 +1,13 @@
+
 ### .setWidgetId()
+<strong class="depracated">DEPRACATED</strong> as of March 12th 2018.  See [switchWidget()](#switchwidget) for improved
```
Comment (sejanick) [inline source/includes/_setwidgetid.md]: "DEPRECATED" is misspelled here.  There also seems to be a word missing at the end of the sentence here: "...See for improved" what?

File: source/includes/_setwidgetid.md [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 
 ```javascript
 var widgetId = '1234-5678-91011-1213-1415'
 SnapEngage.setWidgetId(widgetId);
 ```
 
-Manually set the currently-active widget. A common use case would be offering chat buttons which are linked to separate widgets on a single page, for instance, creating one link that goes to your Help Desk widget, and another that goes to your Sales Team widget. <a href="https://help.snapengage.com/i-would-like-to-offer-chat-with-my-support-and-sales-teams-which-are-in-2-different-snapengage-widgets/">Learn how to set that up right here.</a>
+Manually set the currently-active widget. A common use case would be offering chat buttons which are linked to separate widgets on a single page, for instance, creating one link that goes to your Help Desk widget, and another that goes to your Sales Team widget. <a href="https://help.snapengage.com/i-would-like-to-offer-chat-with-my-support-and-sales-teams-which-are-in-2-different-snapengage-widgets/" target="_blank">Learn how to set that up right here.</a>
```
Comment (sejanick) [inline source/includes/_setwidgetid.md]: Need a colon or a semicolon after "on a single page", like so:

"linked to separate widgets on a single page: for instance, creating one link..."

File: source/includes/_switchwidget.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+var switchCallback = function() {
+    // do something
+};
+
+SnapEngage.switchWidget(newWidget, switchCallback);
+```
+
+Enable switching between two widgets.
+
+We have deprecated `setWidgetId()` API to provide a better solution for switching between two widgets and returning a more accurate status to help with your business needs. Additionally, we now offer an optional callback parameter so you can do more when the new status is returned. Here we do not differentiate between online and offline status, however, you can use the getAgentStatusAsync API to call different methods based on the new status.
```
Comment (sejanick) [inline source/includes/_switchwidget.md]: I thought we didn't end up doing the callback for this?  Is the statement "Additionally, we now offer an optional callback parameter so you can do more when the new status is returned." still accurate?

Comment (sejanick) [reply source/includes/_switchwidget.md]: ...Or was it just the event we didn't do?

Comment (sejanick) [reply source/includes/_switchwidget.md]: Also, I think you're missing a "the" at the beginning of this sentence:

"We have deprecated the `setWidgetId()` API..."

**Comment by lucksp** [comment]
updated with grammar/typo changes.
We did keep the 2nd callback because we the option to do something when the new status comes back.  The code is simplified for now while we try and understand what our customers will do with this API.

**Pull request merged**